drivers/flash/flash_stm32.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2017 BayLibre, SAS.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include <flash_stm32.h>
+
+#if defined(CONFIG_SOC_SERIES_STM32L4X)
+#define FLASH_KEY1	((uint32_t)0x45670123U) /* Flash key1 */
+#define FLASH_KEY2	((uint32_t)0xCDEF89ABU)	/* Flash key2 */
```

Comment:
Thanks, I rebased the F3x merge code on this updated PR: https://github.com/superna9999/zephyr/tree/flash_stm32f3x_merge
if you want to test on a F3.